security: TestUseCerts failed

security.TestUseCerts [failed](https://tanzanite.cluster.engflow.com/invocations/default/92a85e5f-e255-4c0c-b677-51c6a3b77624?testReportRun=8&testReportShard=1&testReportAttempt=1#targets-Ly9wa2cvc2VjdXJpdHk6c2VjdXJpdHlfdGVzdA==) on master @ [a05a56ec16db34644361b2328d889179c3e410c3](https://github.com/cockroachdb/cockroach/commits/a05a56ec16db34644361b2328d889179c3e410c3):


```
        	GOROOT/src/os/file.go:809 +0x158
        crypto/x509.loadSystemRoots()
        	GOROOT/src/crypto/x509/root_unix.go:70 +0x38d
        crypto/x509.initSystemRoots()
        	GOROOT/src/crypto/x509/root.go:40 +0x56
        sync.(*Once).doSlow(0xc004535430?, 0x44dd45?)
        	GOROOT/src/sync/once.go:76 +0xb4
        sync.(*Once).Do(...)
        	GOROOT/src/sync/once.go:67
        crypto/x509.systemRootsPool()
        	GOROOT/src/crypto/x509/root.go:31 +0x45
        crypto/x509.(*Certificate).Verify(0xc002adf608, {{0xc0035374e0, 0x9}, 0xc0048ecc90, 0x0, {0xc1eac023f6088f6f, 0x1d11f41d3, 0x9e1d7a0}, {0x0, 0x0, ...}, ...})
        	GOROOT/src/crypto/x509/verify.go:792 +0x15d
        crypto/tls.(*Conn).verifyServerCertificate(0xc004874a88, {0xc00472cf90, 0x1, 0x1})
        	GOROOT/src/crypto/tls/handshake_client.go:1128 +0xaa5
        crypto/tls.(*clientHandshakeStateTLS13).readServerCertificate(0xc004535c50)
        	GOROOT/src/crypto/tls/handshake_client_tls13.go:665 +0x26f
        crypto/tls.(*clientHandshakeStateTLS13).handshake(0xc004535c50)
        	GOROOT/src/crypto/tls/handshake_client_tls13.go:142 +0x756
        crypto/tls.(*Conn).clientHandshake(0xc004874a88, {0x6413a20, 0xc004876e60})
        	GOROOT/src/crypto/tls/handshake_client.go:366 +0x845
        crypto/tls.(*Conn).handshakeContext(0xc004874a88, {0x6413a20, 0xc004876d70})
        	GOROOT/src/crypto/tls/conn.go:1568 +0x3a6
        crypto/tls.(*Conn).HandshakeContext(...)
        	GOROOT/src/crypto/tls/conn.go:1508
        net/http.(*persistConn).addTLS.func2()
        	GOROOT/src/net/http/transport.go:1651 +0x6e
        created by net/http.(*persistConn).addTLS in goroutine 12391
        	GOROOT/src/net/http/transport.go:1647 +0x309
        goroutine 12394 [select]:
        crypto/tls.(*Conn).handshakeContext.func2()
        	GOROOT/src/crypto/tls/conn.go:1544 +0x86
        created by crypto/tls.(*Conn).handshakeContext in goroutine 12393
        	GOROOT/src/crypto/tls/conn.go:1543 +0x285
        goroutine 12391 [chan receive]:
        net/http.(*persistConn).addTLS(0xc0048cad80, {0x6413a20, 0xc004876d70}, {0xc0035374e0, 0x9}, 0x0)
        	GOROOT/src/net/http/transport.go:1657 +0x327
        net/http.(*Transport).dialConn(0xc00670f680, {0x6413a20, 0xc004876d70}, {{}, 0x0, {0xc0013154c0, 0x5}, {0xc0035374e0, 0xf}, 0x0})
        	GOROOT/src/net/http/transport.go:1738 +0x99c
        net/http.(*Transport).dialConnFor(0xc00670f680, 0xc004ca73f0)
        	GOROOT/src/net/http/transport.go:1563 +0xb8
        net/http.(*Transport).startDialConnForLocked.func1()
        	GOROOT/src/net/http/transport.go:1545 +0x35
        created by net/http.(*Transport).startDialConnForLocked in goroutine 10363
        	GOROOT/src/net/http/transport.go:1544 +0x117: Leaked goroutine: goroutine 12394 [select]:
        crypto/tls.(*Conn).handshakeContext.func2()
        	GOROOT/src/crypto/tls/conn.go:1544 +0x86
        created by crypto/tls.(*Conn).handshakeContext in goroutine 12393
        	GOROOT/src/crypto/tls/conn.go:1543 +0x285
--- FAIL: TestUseCerts (16.54s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=8</code>
 - <code>shard=1</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/security-engineering @cockroachdb/server @cockroachdb/product-security
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUseCerts.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-48341

--- a/pkg/util/leaktest/leaktest.go
+++ b/pkg/util/leaktest/leaktest.go
@@ -63,6 +63,7 @@ func interestingGoroutines() map[int64]string {
 			// TODO(pritesh-lahoti): Revisit this once Go is updated to 1.23, as this seems to have been
 			// fixed: https://github.com/golang/go/pull/62227.
 			strings.Contains(stack, "net/http.(*persistConn).addTLS") ||
+			strings.Contains(stack, "crypto/tls.(*Conn).handshakeContext") ||
 			// Seems to be gccgo specific.
 			(runtime.Compiler == "gccgo" && strings.Contains(stack, "testing.T.Parallel")) ||
 			// Ignore intentionally long-running logging goroutines that live for the

